core/src/main/resources/reference.conf

```diff
@@ -122,14 +122,25 @@ datastax-java-driver {
 
   # The policy that decides the "query plan" for each query; that is, which nodes to try as
   # coordinators, and in which order.
+  # It can be overridden in a profile. Note that the driver creates as few instances as possible:
+  # if a named profile inherits from the default profile, or if two sibling profiles have the
+  # exact same configuration, they will share a single policy instance at runtime.
+  #
+  # If there are multiple load balancing policies in a single driver instance, they work together in
+  # the following way:
+  # - each request gets a query plan from its profile's policy (or the default policy if the request
+  #   has no profile, or the profile does not override the policy).
+  # - when the policies assign distances to nodes, the driver uses the smallest assigned distance
+  #   for any given node.
   load-balancing-policy {
     # The class of the policy. If it is not qualified, the driver assumes that it resides in the
     # package com.datastax.oss.driver.internal.core.loadbalancing.
     #
     # The driver provides a single implementation out of the box: DefaultLoadBalancingPolicy.
     #
     # You can also specify a custom class that implements LoadBalancingPolicy and has a public
-    # constructor with a DriverContext argument.
+    # constructor with two arguments: the DriverContext and a String representing the profile
```

Comment:
I was thinking at first that requiring this constructor signature was be too restrictive...however you could add additional settings to a a profile or context, so it makes sense.